### PR TITLE
conda v2 plugin: support for more architectures

### PR DESCRIPTION
**Pull request by sergiusens**
Using the same strategy as the npm v2 plugin to download the correct
miniconda with the additional testing.

Tests were also migrated to pytest.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/plugins/v2/conda.py [hunk 1]
```diff
@@ -35,15 +35,56 @@
...
+    if snap_arch is not None:
+        try:
+            miniconda_arch = _MINICONDA_ARCH_FROM_SNAP_ARCH[snap_arch]
+        except KeyError:
+            raise ArchitectureMissing(snap_arch)
+    # But there may be times when running from a virtualenv while doing development.
+    else:
+        miniconda_arch = _MINICONDA_ARCH_FROM_PLATFORM[platform.machine()][
+            platform.architecture()[0]
+        ]
```
Comment (cmatsuoka) [inline snapcraft/plugins/v2/conda.py:79]: Is x86_64 the only possible machine here? Otherwise error handling is welcome.

**Review by cmatsuoka — COMMENTED** [review]
Looks good, added a question about

File: snapcraft/plugins/v2/conda.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/plugins/v2/conda.py:79]: Yes, and only applies to development environment, we have a similar strategy in place for our npm plugin

Comment (cjp256) [reply snapcraft/plugins/v2/conda.py:79]: Considering the PR is to support more architectures... :thinking: lol

File: snapcraft/plugins/v2/conda.py [hunk 2]
```diff
@@ -35,15 +35,56 @@
...
+}
+_MINICONDA_ARCH_FROM_PLATFORM = {"x86_64": {"32bit": "x86", "64bit": "x86_64"}}
+
+
+class ArchitectureMissing(SnapcraftException):
+    def __init__(self, snap_arch: str) -> None:
+        self.snap_arch = snap_arch
+
+    def get_brief(self) -> str:
+        return f"Architecture {self.snap_arch!r} is not supported."
```
Comment (cjp256) [inline snapcraft/plugins/v2/conda.py:61]:
```suggestion
        return f"Architecture {self.snap_arch!r} is not supported for the 'conda' plugin."
```

**Review by cmatsuoka — APPROVED** [review]

**Review by cjp256 — APPROVED** [review]

**Pull request merged**